pkg/costmodel/router.go

```diff
@@ -692,8 +692,15 @@ func (a *Accesses) PrometheusQuery(w http.ResponseWriter, r *http.Request, _ htt
 		return
 	}
 
+	// TODO test to make sure "time" does not get set, if not given
+
+	var ts time.Time
+	if qp.GetInt64("time", 0) > 0 {
+		ts = time.Unix(qp.GetInt64("time", 0), 0)
+	}
```

Comment:
Mainly want to make sure that this is cool with @mbolt35

Reply:
Was very careful with the logic, but I don't want to mess anything up here.

Reply:
This all looks great to me, check the comment in query.go

Reply:
If we want to match the Prom query [1] maybe we should also accept an RFC3339 timestamp.

[1] https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries

Reply:
Also, do you think we should error if a negative number comes in? I don't think its a huge deal here, but we've definitely had cases where silently overwriting a user-passed parameter because it was "out of bounds" caused confusing behavior.

Reply:
Good points

Reply:
Updated!